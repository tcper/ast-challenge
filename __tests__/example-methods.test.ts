import * as ts from "typescript";
import { readFileSync, writeFileSync } from 'fs';
import generateFunction from '../src/index'

const data = JSON.parse(readFileSync(__dirname + '/../example-methods.json', 'utf-8'));

it('works', () => {
    Object.keys(data).forEach(key => {
        const input = data[key];
        let ast = generateFunction(input);
        const printer = ts.createPrinter({ newLine: ts.NewLineKind.LineFeed });
        let code = ast.map((line) => printer.printNode(ts.EmitHint.Unspecified, line, ts.createSourceFile("source.ts", "", ts.ScriptTarget.ESNext, false, ts.ScriptKind.TS))).join("\n")
        expect(code).toMatchSnapshot()
    })
});