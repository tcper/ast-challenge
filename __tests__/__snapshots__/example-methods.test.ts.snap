// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolsResponse, TData> {
    request?: QueryPoolsRequest;
}
const usePools = <TData = QueryPoolsResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryPoolsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 2`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryNumPoolsResponse, TData> {
    request?: QueryNumPoolsRequest;
}
const usePools = <TData = QueryNumPoolsResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryNumPoolsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 3`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryTotalLiquidityResponse, TData> {
    request?: QueryTotalLiquidityRequest;
}
const usePools = <TData = QueryTotalLiquidityResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryTotalLiquidityResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 4`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolResponse, TData> {
    request?: QueryPoolRequest;
}
const usePools = <TData = QueryPoolResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryPoolResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 5`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolParamsResponse, TData> {
    request?: QueryPoolParamsRequest;
}
const usePools = <TData = QueryPoolParamsResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryPoolParamsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 6`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryTotalPoolLiquidityResponse, TData> {
    request?: QueryTotalPoolLiquidityRequest;
}
const usePools = <TData = QueryTotalPoolLiquidityResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryTotalPoolLiquidityResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 7`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryTotalSharesResponse, TData> {
    request?: QueryTotalSharesRequest;
}
const usePools = <TData = QueryTotalSharesResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QueryTotalSharesResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 8`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QuerySpotPriceResponse, TData> {
    request?: QuerySpotPriceRequest;
}
const usePools = <TData = QuerySpotPriceResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QuerySpotPriceResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 9`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QuerySwapExactAmountInResponse, TData> {
    request?: QuerySwapExactAmountInRequest;
}
const usePools = <TData = QuerySwapExactAmountInResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QuerySwapExactAmountInResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;

exports[`works 10`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QuerySwapExactAmountOutResponse, TData> {
    request?: QuerySwapExactAmountOutRequest;
}
const usePools = <TData = QuerySwapExactAmountOutResponse>({ request, options }: UsePoolsQuery<TData>) => {
    return useQuery<QuerySwapExactAmountOutResponse, Error, TData>([\\"poolsQuery\\", request], () => {
        if (!queryService)
            throw new Error(\\"Query Service not initialized\\");
        return queryService.pools(request);
    }, options);
};"
`;
